.husky/pre-commit

```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+echo "Running PRE-COMMIT hook"
+
+npx lint-staged
+npm dedupe
+
+echo "Don't for get to commit your deduped package-lock.json!"
```

Comment:
What is the reason of this comment/log?

It `package-lock.json` should not be committed, we could run `git checkout package-lock.json`.

However, I am not sure it shouldn’t be committed. I’d actually go further and made sure the lock file is up to date with `package.json` and force-re-install the packages ignoring the lock file and committing the lock file if there are any changes.

WDYT? 🤔

Reply:
no, this is saying it _should_ be committed, because `npm dedupe` ensures the package-lock.json is deduped and minimized to decrease the installs necessary. I was debating also immediately `git add`'ing package-lock.json to accomplish this, but i felt this would be inconsistent with how lint-staged will modify files but also not add them to git.

Reply:
Then, why don’t we actually check it is committed? For example using this:

```sh
if [ -f package-lock.json ]; then
  echo "Don't for get to commit your deduped package-lock.json!"
  exit 2
fi
```

Or even better: why don’t we first install the packages (thus the lock file is created), dedupe it and commit the lock file if it has any meaningful changes? :thinking:

> i felt this would be inconsistent with how lint-staged will modify files but also not add them to git

I have never used `lint-staged`, therefore, that is up to you, however, it might be a good idea to commit the changed lock file if there are any meaninful changes.

Reply:
ok, but this is in husky, meaning it's on the developer side before the commit takes place. the lock file is already created, dedupe requires that packages are already installed, or it would fail. i don't feel the need to do _everything_ for the developer, contributors should be expected to at least set up the project if they're doing anything that would change either the node modules in package.json or package-lock.json

Reply:
I'm still looking at the right way to potentially add anything staged so the experience is that the fixes "just happens" and we don't have to worry about committing twice or readding files or anything like that

Reply:
i say that, then i find that lint-staged automatically handles all of that, so it'd just be for the dedupe part that i wanna sort it out

Reply:
> I'm still looking at the right way to potentially add anything staged so the experience is that the fixes "just happens" and we don't have to worry about committing twice or readding files or anything like that

I don’t like the idea to stage anything automatically, as I usually have multiple files changed, but I might commit them separately.

Reply:
yeah, this only stages those files you've already included in your commit (from lint-staged), and the only one i'm looking to add is package-lock to be automatically staged, but as-is, we're only linting the files you've already staged

Reply:
As I have already said, I disagree with staging any files automatically. I might have some unstaged changes to `package*.json` files, however, I might want commit those changes in a later commit.

Reply:
right, what i'm saying is i'm not changing that yet, and given your input, i won't automatically add package*.json

Reply:
I am considering, that said, requiring a "dedupe" check after the test that would fail CI if your package-lock isn't deduped so it would need to be deduped before the PR could be merged

Reply:
i've got that part added